app/features/welcome/components/Welcome.js

```diff
@@ -0,0 +1,145 @@
+    align-items: center;
+    justify-content: center;
+`;
+
+const WelcomeWrapper = CenterContent.extend`
+    background: linear-gradient(#165ecc,#44A5FF);
+    height: 100vh;
+    width: 100%;
+`;
+
+// $FlowFixMe
+import { FieldTextStateless } from '@atlaskit/field-text';
+
+// $FlowFixMe
+import Button from '@atlaskit/button';
+
+// $FlowFixMe
+import { AtlasKitThemeProvider } from '@atlaskit/theme';
+
+
+type Props = {
+
+    /**
+     * React Router history object (for flow)
+     */
+    history: Object;
+};
+
+type State = {
+
+    /**
+     * URL of the room to join
+     * If this is not a url it will be treated as room name for default domain
+     */
+    url: string;
+};
+
+/**
+ * Welcome Component.
+ */
+export default class Welcome extends Component<Props, State> {
+
+    /**
+     * Initialize Welcome Component
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            url: ''
+        };
+
+        // Bind event handlers
+
+        // $FlowFixMe
+        this._onURLChange = this._onURLChange.bind(this);
+
+        // $FlowFixMe
+        this._onFormSubmit = this._onFormSubmit.bind(this);
+
+        // $FlowFixMe
+        this._onJoin = this._onJoin.bind(this);
+    }
+
+    /**
+     * Render function of component.
+     *
+     * @return {ReactElement}
+     */
+    render() {
+        return (
+            <AtlasKitThemeProvider mode='light'>
+                <WelcomeWrapper>
+                    <CenterContent>
+                        <form onSubmit = { this._onFormSubmit }>
+                            <FieldTextStateless
+                                autoFocus = {true}
+                                isLabelHidden = {true}
+                                shouldFitContainer = { true }
+                                onChange = { this._onURLChange }
+                                type = 'text'
+                                value = { this.state.url } />
+                        </form>
+                        <Button
+                            appearance='primary'
+                            type='button'
+                            onClick={ this._onJoin }>
+                            GO
+                        </Button>
+                    </CenterContent>
+                </WelcomeWrapper>
+            </AtlasKitThemeProvider>
+        );
+    }
+
+    /**
+     * Keeps URL input value and URL in state in sync
+     */
+    _onURLChange(event: Event) {
+        this.setState({
+
+            // $FlowFixMe
+            url: event.target.value
+        });
+    }
+
+    /**
+     * Prevents submission of the form and delagates join logic.
+     */
+    _onFormSubmit(event: Event) {
+        event.preventDefault();
+        this._onJoin();
+    }
+
+    /**
+     * Redirect and join conference
+     */
+    _onJoin() {
+
+        const url = URL.parse(this.state.url);
+
+        // Check if the parsed url is a full url or just room name
+        if (url.host) {
```

Comment:
You also need to check if there is a path different than `/` and not do anything in that case.

Reply:
If path is `/` then it redirects to `/` so stays on the same page.

Reply:
Yes, but I think we shouldn't enable the button when the URL is invalid. Also, I guess we should check if the scheme is http or https, otherwise it's not a valid URL!

Reply:
What if the user doesn't enter protocol?
Let's say it enters 'beta.meet.jit.si/room'. Shouldn't it go to `beta.meet.jit.si` server and join room `room`??

Reply:
For consistency with the mobile app: nope. I know it makes sense to try to do that, so maybe we can revisit this in the future.